src/screens/nano/NanoContractExecuteMethod.js

```diff
@@ -0,0 +1,635 @@
+        continue;
+      }
+
+      // All other types we expect as strings
+      argValues.push(value);
+    }
+
+    ncData.args = argValues;
+
+    // Then we get the list of actions
+    const actionsData = [];
+    for (const action of actions) {
+      if (!action.type) {
+        // We will skip if the user has just added an empty action
+        continue;
+      }
+
+      const amountValue = isNFT(action.token) ? action.amount : walletUtils.decimalToInteger(action.amount, decimalPlaces);
+      action.amount = amountValue;
+
+      actionsData.push(action);
+    }
+
+    ncData.actions = actionsData;
+
+    return await wallet.createNanoContractTransaction(data.method, address, ncData, { pinCode: pin });
+  }
+
+  /**
+   * Return if token is an NFT
+   *
+   * @param {string} token Token uid to check if it's NFT
+   */
+  const isNFT = (token) => {
+    return helpers.isTokenNFT(token, tokenMetadata);
+  }
+
+  /**
+   * This renders all the options of the token select for a tokenUid argument
+   */
+  const renderTokenOptions = () => {
+    return registeredTokens.map((token) => {
+      return <option value={token.uid} key={token.uid}>{token.name} ({token.symbol})</option>;
+    })
+  }
+
+  /**
+   * If an argument of the method is a token uid, we show a select
+   * with all tokens registered in the wallet
+   *
+   * @param {Object} ref React reference object
+   * @param {boolean} required If this argument is required
+   * @param {Function} onChange Method called when the select changes
+   */
+  const renderTokenSelect = (ref, required, onChange) => {
+    return (
+      <select ref={ref} required={required} className="form-control" onChange={onChange}>
+        <option value='' key=''> - </option>
+        {renderTokenOptions()}
+      </select>
+    );
+  }
+
+  /**
+   * Method that has the logic to render the input for each argument
+   *
+   * Amount argument is an InputNumber
+   * TokenUid will show a select with registered tokens
+   * int and float are simple numbers
+   * Timestamp will have a datetime-local
+   * The rest will be a text input
+   *
+   * @param {string} type Argument type to render the input
+   * @param {boolean} isOptional If this argument is optional
+   * @param {Object} ref React reference object
+   */
+  const renderInput = (type, isOptional, ref) => {
+    if (type === 'Amount') {
+      return <InputNumber
+              required={!isOptional}
+              requirePositive={true}
+              ref={ref}
+              className="form-control output-value"
+              placeholder={hathorLib.numberUtils.prettyValue(0, decimalPlaces)}
+            />
+    }
+
+    if (type === 'TokenUid') {
+      return renderTokenSelect(ref, !isOptional);
+    }
+
+    let inputType;
+    if (type === 'int' || type === 'float') {
+      inputType = 'number';
+    } else if (type === 'Timestamp') {
+      inputType = 'datetime-local';
+    } else {
+      inputType = 'text';
+    }
```

Comment:
This is a suggestion, but if you prefer the if-else approach just ignore this.

```suggestion
    switch(type) {
      case 'int':
      case 'float':
        inputType = 'number';
        break;
      case 'Timestamp':
        inputType = 'datetime-local';
        break;
      default:
        inputType = 'text';
    }
```

Reply:
I like it, done https://github.com/HathorNetwork/hathor-wallet/pull/627/commits/71be65d9dbc5a0c2d9ceb66f1b35f0fa55a4661f